### PR TITLE
Miscellaneous static bug analysis fixes.

### PR DESCRIPTION
**Pull request by johnou**

### CODE REVIEW

File: storage/src/main/java/io/atomix/storage/buffer/FileBytes.java [hunk 1]
```diff
@@ -249,15 +249,15 @@ public Bytes read(int position, Bytes bytes, int offset, int length) {
     if (bytes.hasArray()) {
       try {
         seekToOffset(position);
-        randomAccessFile.read(bytes.array(), (int) offset, (int) length);
+        randomAccessFile.readFully(bytes.array(), (int) offset, (int) length);
```
Comment (johnou) [inline storage/src/main/java/io/atomix/storage/buffer/FileBytes.java]: Although RandomAccessFile is not a subclass of InputStream, this method behaves in exactly the same way as the InputStream.read() method of InputStream.
https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#read(byte[],%20int,%20int)
An attempt is made to read as many as len bytes, but a smaller number may be read.

File: core/src/test/java/io/atomix/core/AtomixTest.java [hunk 2]
```diff
@@ -230,6 +231,7 @@ public void onEvent(ClusterEvent event) {
       try {
         queue.put(event);
       } catch (InterruptedException e) {
+        Throwables.propagate(e);
```
Comment (kuujo) [inline core/src/test/java/io/atomix/core/AtomixTest.java]: Isn’t this deprecated now? Not sure what’s supposed to replace it though.

File: core/src/test/java/io/atomix/core/tree/impl/DocumentTreeTest.java [hunk 3]
```diff
@@ -300,7 +300,7 @@ public void testClear() throws Exception {
   /**
    * Tests listeners.
    */
-  @Test
+  @Test(timeout = 45000)
```
Comment (kuujo) [inline core/src/test/java/io/atomix/core/tree/impl/DocumentTreeTest.java]: Can we just create a base class that assigns a timeout to all the tests?

File: core/src/test/java/io/atomix/core/AtomixTest.java (hunk 2 above)
Comment (johnou) [reply core/src/test/java/io/atomix/core/AtomixTest.java]: Yep, in this cause I would probably just interrupt the thread again, but instead copied code which is like this in about three places..

File: core/src/test/java/io/atomix/core/tree/impl/DocumentTreeTest.java (hunk 3 above)
Comment (johnou) [reply core/src/test/java/io/atomix/core/tree/impl/DocumentTreeTest.java]: Not to first look, only thing I see is per class at suite level.

File: core/src/test/java/io/atomix/core/AtomixTest.java (hunk 2 above)
Comment (johnou) [reply core/src/test/java/io/atomix/core/AtomixTest.java]: updated.

File: core/src/test/java/io/atomix/core/tree/impl/DocumentTreeTest.java (hunk 3 above)
Comment (johnou) [reply core/src/test/java/io/atomix/core/tree/impl/DocumentTreeTest.java]: This guy is the main suspect though I was able to reproduce the never ending test like in Travis.

**Pull request merged**